refactor(fivetran_sdk): Restructure the fivetran_sdk repo (#88)
fivetran/fivetran_sdk: README.md

# Fivetran Partner SDK
A way for partners to create source and destination connectors that run on [Fivetran’s platform](https://www.fivetran.com/). Read more in our [product docs](https://fivetran.com/docs/partner-built-program). 

## Repo Structure
This repo consists of example source and destination connectors along with a local testing environment. Both connectors are written in a [gRPC supported language](https://grpc.io/docs/languages/) that can generate a statically linked binary. We recommend Java, Golang, or Rust. 

> * [V2 Examples](v2/examples)
> * [V1 Examples](v1/examples)

Reviewer: ```suggestion
* [V1 Examples](v1/examples)
* [V2 Examples](v2/examples)
```

V1 then V2 makes more sense from a list point of view

Author: V2 examples are the latest examples, we would want the partner to go through these examples now, instead of V1, so added this before V1.

* [Local Testing Tools](tools/)

## Development
You can follow the [SDK Development Guide](development-guide.md) for guidance on how to develop your code. 

> > Note: New partners should use [V2 protos](v2), while only old partners should continue using [V1 protos](v1).

Reviewer 2: This needs rewording. @5tran-alexil can you suggest?

Author: I have pinged @fivetran-dejantucakov regarding the documentation change.

Reviewer: ```suggestion
> NOTE: If you're a new partner that just started working with the Connector SDK, use [V2 protos](v2). We will continue to support [V1 protos](v1) for partners that have used them to build their SDK connectors.
```

Once you have your code ready to run:
1. Start up your connector running on port 50051 (for destination code, use port 50052).
2. Run the local test environment.

Our READMEs for [source connectors](tools/source-connector-tester/) and [destination connectors](tools/destination-connector-tester/) will take you through setting up the local test environment. 

> > To run an example connector from `examples` directory, follow the README specific to the example. 

Reviewer: ```suggestion
> NOTE: To run an example connector from the `examples` directory, follow the README specific to the example. 
```

## Partner Process

### How does it work?
You build a source or destination connector using our SDK in this repo. We’ll deploy it to our platform, and customers can discover it via the Fivetran website, dashboard, and product documentation. We’ll tag it with a `partner-built` designation and explain the support process to customers. 

### How do we deploy?
Right now, we are employing manual processes before getting into automation:
1. You put your code in a public GitHub repo or provide a couple of our team members access to your private repo. 
2. We pull the code, inspect it, and build the executable binary to deploy to our platform. Allow up to a week for us to complete our review and deploy process.

### How do we test?
We expect testing to ensure your source or destination connector is working as expected, either with test data or with a customer who is eager to adopt. We provide [testers](tools/) in this repo you can use locally without connecting to Fivetran platform. We also recommend you think through typical scenarios for testing:

* Handle all data type mappings (between Fivetran data types and source/destination data types)

> * Big data loads

Reviewer 2: Line 52 has 'connector' need to change that to source
